src/envo/envo-edit.owl

```diff
@@ -28119,6 +28142,155 @@ SubClassOf(<http://purl.obolibrary.org/obo/ENVO_03000032> <http://purl.obolibrar
+SubClassOf(<http://purl.obolibrary.org/obo/ENVO_03000044> <http://purl.obolibrary.org/obo/ENVO_03000043>)
+SubClassOf(<http://purl.obolibrary.org/obo/ENVO_03000044> ObjectSomeValuesFrom(<http://purl.obolibrary.org/obo/RO_0000057> <http://purl.obolibrary.org/obo/ENVO_00002149>))
+SubClassOf(<http://purl.obolibrary.org/obo/ENVO_03000044> ObjectSomeValuesFrom(<http://purl.obolibrary.org/obo/RO_0002234> <http://purl.obolibrary.org/obo/ENVO_00002200>))
+
+# Class: <http://purl.obolibrary.org/obo/ENVO_03000045> (brine rejection process)
+
+AnnotationAssertion(Annotation(<http://www.geneontology.org/formats/oboInOwl#hasDbXref> "https://en.wikipedia.org/wiki/Brine_rejection") <http://purl.obolibrary.org/obo/IAO_0000115> <http://purl.obolibrary.org/obo/ENVO_03000045> "A material transformation process during which brine is formed due to the expulsion of salt during frazil formation or a seawater freezing process. Expelled salt accumulates in brine droplets, which remain in a liquid state due to their increased salinity.")
+AnnotationAssertion(<http://purl.obolibrary.org/obo/IAO_0000117> <http://purl.obolibrary.org/obo/ENVO_03000045> "http://orcid.org/0000-0002-3410-4655")
+AnnotationAssertion(rdfs:label <http://purl.obolibrary.org/obo/ENVO_03000045> "brine rejection process"@en)
+SubClassOf(<http://purl.obolibrary.org/obo/ENVO_03000045> <http://purl.obolibrary.org/obo/ENVO_03000043>)
+SubClassOf(<http://purl.obolibrary.org/obo/ENVO_03000045> ObjectSomeValuesFrom(<http://purl.obolibrary.org/obo/RO_0000057> <http://purl.obolibrary.org/obo/ENVO_00002149>))
+SubClassOf(<http://purl.obolibrary.org/obo/ENVO_03000045> ObjectSomeValuesFrom(<http://purl.obolibrary.org/obo/RO_0002234> <http://purl.obolibrary.org/obo/ENVO_00003044>))
+
+# Class: <http://purl.obolibrary.org/obo/ENVO_03000046> (frazil)
+
+AnnotationAssertion(Annotation(<http://www.geneontology.org/formats/oboInOwl#hasDbXref> "https://nsidc.org/cryosphere/glossary/term/frazil") <http://purl.obolibrary.org/obo/IAO_0000115> <http://purl.obolibrary.org/obo/ENVO_03000046> "Small needle-like ice crystals consisting of nearly pure fresh water, which are suspended in water.")
+AnnotationAssertion(<http://purl.obolibrary.org/obo/IAO_0000117> <http://purl.obolibrary.org/obo/ENVO_03000046> "http://orcid.org/0000-0002-3410-4655")
+AnnotationAssertion(rdfs:label <http://purl.obolibrary.org/obo/ENVO_03000046> "frazil"@en)
+SubClassOf(<http://purl.obolibrary.org/obo/ENVO_03000046> <http://purl.obolibrary.org/obo/ENVO_01000846>)
+
+# Class: <http://purl.obolibrary.org/obo/ENVO_03000047> (frazil formation)
+
+AnnotationAssertion(Annotation(<http://www.geneontology.org/formats/oboInOwl#hasDbXref> "https://nsidc.org/cryosphere/glossary/term/frazil") <http://purl.obolibrary.org/obo/IAO_0000115> <http://purl.obolibrary.org/obo/ENVO_03000047> "A material accumulation process in which frazil ice crystals merge, forming thin sheets of ice.")
```

Comment:
> frazil ice formation: def = An ice formation process in which frazil ice crystals form due to the freezing of open, turbulent, supercooled water.
> 
> with subclass axioms:
> 
> 'has input' some 'water mass'
> 'has output' some 'frazil ice'

The input should probably be `water` (a material), otherwise it's looking good.

> We now need a new set of classes to describe the various steps in the transition processes between frozen water and sea ice as subclasses of a sea ice formation process Here are additional classes to be added in order to encapsulate the process of sea ice formation, as well as some other important ice terms:

Wow! this is enough for another PR, but let's give it a shot:
 
> The overall process of sea ice formation proceeds as a continuum from seawater -> new ice -> nilas -> young ice -> first year ice -> second year ice -> multiyear ice.

Makes sense. This may be a place for some of the causal relations from RO.


> Within the occurrent hierarchy, I'll add:
> 
> material transformation process:
> 
> ice formation process def = A material transformation process during which water is frozen into water ice.

Would we not include depositional processes here? That is, water vapour deposited as water ice?

> -- new ice formation process def = An ice formation process during which water is frozen into a form of new ice such as frazil, grease ice, slush, or shuga.

This could then be a union class of 3 processes:
* frazil formation process
* grease ice formation process
* slush formation process
* shuga formation process

You can make an equivalence axiom like
`'ice formation process' and 'has output' some ('frazil ice' or 'grease ice' or 'slush ice' or 'shuga')


> -- sea ice formation process def = An ice formation process during which seawater is frozen into a sea ice floe.

Hmm, does it have to be a sea ice floe? I'd prefer that the output is 'sea ice' as it may not be in the form of a floe at the early stages.


> --- nilas formation process  def = A sea ice formation process during which new ice such as frazil ice crystals freeze together, forming nilas, a thin, elastic, continuous, ice sheet of a transparent gray color.

We could make a union class for `new ice` and then remove the "such as' bit from this def. That means we'd have an equivalence axiom for `new ice`:
```
ice and 'composed primarily of' some ('frazil ice' or 'grease ice' or 'slush ice' or shuga)
```

We could then add the following subclass axioms for `nilas formation process`:
```
'has input' some 'new ice' 
'has output' some nilas
```

> --- young ice formation process def = A sea ice formation process during which water freezes to the bottom of an existing nilas ice floe creating a thicker ice floe known as young ice.

Is there a clear threshold where nilas becomes new ice? When it loses its transparency?
Note that PATO has [`transparent`](http://purl.obolibrary.org/obo/PATO_0000964), please add this to our imports if its not there already.

> --- first year ice formation process def = A sea ice formation process during which seawater freezes onto young ice during the cold season.
> 
> comment: first year ice does not survive the spring and summer months due to melting.

Not sure about the comment here - surely some first year ice survives to be second year ice? the definition sounds more appropriate for "one year ice" and even then, it doesn't survive the whole year, just the cold seasons. Perhaps "seasonal ice" is a more appropriate label for the ice type this process refers to.

first year ice would be ice that endures for one complete seasonal cycle of a planet or similar astronomical body.


> --- second year ice formation: def = A sea ice formation process during which seawater freezes onto first year ice during the cold season, and in which the ice floe has previously survived one melt season.

This would make more sense if the semantics for first year ice are updated as suggested above: if first year ice melts away in spring and summer, second year ice would never form. Once again, I'd leave out the ice floe in this def as this ice type need not be part of one.

> --- multiyear ice formation process def = A sea ice formation process during which seawater freezes onto second year ice during the cold season, and the ice floe has previously survived more then two melt seasons.

The second part of this definition (",and the ice floe...") is not needed as this is bundled in with `second year ice`. 

> Then within the environmental material hierarchy I'll add the following classes:
[...]
> -new ice def = Water ice which is formed from freezing water during a new ice formation process and is suspended in a water body.
>
> Comment: sea ice types include frazil, grease ice, slush, and shuga.

Make sure to axiomatise as suggested above.


> -- shuga  def = Spongy, lumpy and opaque new ice, which forms during a water freezing in an agitated water body.

Rephrase to:
```
Ice which is formed through the freezing of water in an agitated water body, conferring it spongy, lumpy, and opaque properties.
```

> -- frazil ice def = New ice which is composed of small, needle-like crystals consisting of nearly pure fresh water.
> note to myself as editor: I need to add subclass axiom to also make it subset of new ice, as it is currently only a subset of water ice crystal.

It can't be a subclass of `water ice crystal` as that's countable and thus not a material. Also, if we make `new ice` a union class, we don't need to make the various kinds of new ice subclasses (the reasoner will do that). We'd just have them as subclasses of `water ice`.

> --- grease ice def = A thin, layer of frazil ice crystals which clump together.
> 
> comment: Within a marine system, grease ice makes the ocean surface resemble an oil slick.

This is a bit tricky: if grease ice is made up of frazil ice (that is, the frazil ice hasn't been altered) then maybe this isn't a material but a `grease ice layer` (subclass of `layer`).

> ---- slush ice def = Grease ice which is mixed with snow.

We'll get back to this once we figure out grease ice.

> -ice floe def = A cohesive sheet of water ice which floats upon the surface of a water body.
> 
> comment: Ice floes can exist in sea ice, in rivers or lakes.

Axiomatise appropriately with `'composed primarily of' some 'water ice'` etc

> -- sea ice floe def = An ice floe which is formed from frozen sea water, and floats upon the surface of a marine water body

Axiomatise appropriately with `output of` and `composed primarily of` etc

> --- pancake ice def = A sea ice floe which has a rounded or circular shape and raised rims.

Are there any PATO qualities we can use here to express these shapes? If not, we should request them.

> --- nilas def = A thin, elastic, continuous, sea ice floe of a transparent gray color, which results from the freezing of seawater during a nilas formation process.

I'd drop the process part in the def, the process should have  subclass axiom `has output some nilas`. 

Also, transparent is not a colour property, so rephrase that. Again, check PATO for quality classes and request new ones if needed.

> --- young ice def = An inflexible sea ice floe, thicker then nilas which is formed during a young ice formation process.

Again, we can drop the process part of this def. Check PATO for `inflexible` or similar.

> Comment: Young ice is not as flexible as nilas, but tends to break under wave action. In a compression regime, it will either raft (at the gray ice stage) or ridge (at the gray-white ice stage).

Here you contradict the def - if something is inflexible it can't be 'not as flexible' as something else. It would be interesting to have the rafting and ridging processes in the next PR. Perhaps we can create issues for these.

> --- first year ice = A sea ice floe thicker then young ice, which results from a first year ice formation process.

then --> than

Also, while this ice type does occur in floes, I'm not sure that's necessary (it's also not a subclass of sea ice floe if it's in the material hierarchy). 

Here's a good example of where some of the info from the process def would be useful: Incorporate some to give users an idea of what seasonal fluctuations this ice has been subject to.

> Comment: Forms from water freezing on to the bottom of the existing ice sheet, a process called congelation growth. First year ice does not survive a full melt season as it is melted away.

Again, this doesn't make sense or there would be no second year ice. This suggests we need `seasonal ice` and `first year ice` as separate categories as noted above. If existing glossaries don't differentiate between them, we should synonymise and add editor notes explaining why.

> --- second year ice def = A sea ice floe thicker then first year ice, which results from a second year ice formation process.
> 
> comment: Second year ice has survived melting during a full melt season.

This can't be an ice floe, as explained above.
A bit more description in the def would be useful, just noting that it endured one melt season. 
 
> --- multiyear ice def = A sea ice floe thicker then second year ice, which results from a multiyear ice formation process.
> comment: Multiyear ice has survived melting during more then two full melt seasons.

This can't be an ice floe. Same issues with the def as above. The comment material should probably be integrated with the def.